### DISCUSSION TITLE
Putting not serializable objects to `state`

### DISCUSSION THREAD

**Discussion by kerim371** [Kitware/trame#583] [General] [answered]
Hi,

I'm wondering what is the difference between making variable somehow global (declare somwhere at the top of the script) and putting it as the `state` member?

For example I would like to add `state.view = vtk.VtkLocalView(renderWindow)` as `state` member to be able to update the view every time new actor added/removed but I get error:
```
Skip state value for 'view' since its content is not serializable
```
How serious the error? Because I still can access this member variable.

Another option for me is to declare `view = None` somewhere at the top of the script and then use `view.update()` whenever it is needed.

**Comment by jourdain** [comment, marked as answer]
So the main purpose of `server.state` is to share data with the front-end (browser). Therefore it does not make much sense to put something in it that you don't want to send over the network.

Technically you could use `server.context` ([API](https://trame.readthedocs.io/en/latest/core.server.html#trame_server.core.Server.context)) which is a server side only state which aim for you to leverage its reactive nature. Or put global things to it. 

But in your use case, it seems that you only want to keep track of methods so you can easily call them anywhere. Well in that case, you can use the `server.controller` to bind methods to, so you can call them later when ever you need. In all our example that is what we do for keeping track of `reset_camera` and `update`. ([example](https://github.com/Kitware/trame/blob/master/examples/06_vtk/01_SimpleCone/ClassRemoteRendering.py#L102-L103))

**Comment by kerim371** [comment]
Thank you!
That makes sense, good to know.
